magenta-lib/src/main/scala/magenta/deployment_type/SelfDeploy.scala

```diff
@@ -0,0 +1,71 @@
+package magenta.deployment_type
+
+import java.io.File
+
+import magenta.tasks.{ChangeSwitch, S3Upload}
+
+object SelfDeploy extends DeploymentType with S3AclParams {
+  val name = "self-deploy"
+  val documentation =
+    """
+      |This is a special deployment type that can be used to self-deploy Riff-Raff.
+      |
+      |The approach is to upload a file to S3 and then hit a switch on the singleton instance.
+      |When the switch is hit it will shutdown with a specific exit code at the next
+      |opportunity (when it has finished deploys).
+    """.stripMargin
+
+  val bucket = Param[String]("bucket",
+    """
+      |S3 bucket name to upload artifact into.
+      |
+      |The path in the bucket is `<stack>/<stage>/<packageName>/<fileName>`.
+    """.stripMargin
+  )
+
+  val prefixStage = Param[Boolean]("prefixStage",
```

Comment:
When I was reviewing this I thought I might actually get rid of them entirely for this deploy type. I can't see why the values would ever need to be customised.